dao/l1syncedblock/l1_synced_block.go

```diff
@@ -72,6 +75,39 @@ func (m *defaultL1EventModel) DropL1SyncedBlockTable() error {
 	return m.DB.Migrator().DropTable(m.table)
 }
 
+func (m *defaultL1EventModel) GetLatestL1Block() (blockInfo *L1SyncedBlock, err error) {
+	dbTx := m.DB.Table(m.table).Order("id desc").Find(&blockInfo)
+	if dbTx.Error != nil {
+		return nil, types.DbErrSqlOperation
+	}
+	if dbTx.RowsAffected == 0 {
+		return nil, types.DbErrNotFound
+	}
+	return blockInfo, nil
+}
+
+func (m *defaultL1EventModel) GetL1BlockById(id uint) (blockInfo *L1SyncedBlock, err error) {
+	dbTx := m.DB.Table(m.table).Where("id = ?", id).Find(&blockInfo)
+	if dbTx.Error != nil {
+		return nil, types.DbErrSqlOperation
+	}
+	if dbTx.RowsAffected == 0 {
+		return nil, types.DbErrNotFound
+	}
+	return blockInfo, nil
+}
+
+func (m *defaultL1EventModel) DeleteL1Block(blockInfo *L1SyncedBlock) (err error) {
+	dbTx := m.DB.Table(m.table).Unscoped().Delete(&blockInfo) // delete record permanently
```

Comment:
I think we can use `where id < ?` to delete blocks in batch?

Reply:
agree